src/movement/maestro_thruster_controller.cpp

```diff
@@ -0,0 +1,207 @@
+#include "movement/maestro_thruster_controller.hpp"
+namespace rs
+{
+    /**
+     * Constructor.
+     *
+     */
+    MaestroThrusterController::MaestroThrusterController()
+    {
+        _is_initialized = false;
+
+
+    }
+
+
+    /**
+     * Deconstructor.
+     *
+     */
+    MaestroThrusterController::~MaestroThrusterController()
+    {
+    }
+
+    /**
+     * initializes the thruster controller. Must be called before using any
+     * other functions
+     *
+     * @param thrusters The number of thrusters controlled by the controller.
+     * @param max_speed The maximum normalized speed in the forward and reverse
+     *        directions that should be allowed by the controller.
+     * @param delay_ms Delay specified in milliseconds to continue sending
+     *        zeroing signal to continue thruster operation.
+     * @param port The serial port to be used for communicating with the
+     *        thrusters. This port may be changed during runtime.
+     */
+    int MaestroThrusterController::init( const double max_speed,
+                                           Serial *port,
+                                           const double delay_ms)
+    {
+        if(setPort(port) != 0)
+        {
+            return -1;
+        }
+
+        _post_reset_delay_ms == delay_ms;
+
+        //initialze thruster settings
+        //loop over all possible channels (0-11)
+        ros::Time now = ros::Time::now();
+        for(uint8_t i = 0; i < _max_thrusters; ++i)
+        {
+            //initialze reset time
+            _next_reset[i] = now;
+
+            //set max speed for thruster
+            _max_speed[i] = max_speed;
+        }
+
+        if (delay_ms < 185.00)
```

Comment:
Apparently this was found empirically, though I agree it probably should be `#defined` or made a constant.